src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,54 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        FigureType figureType = FigureType.values()[random.nextInt(FigureType.values().length)];
+        Figure figure = null;
+        String color = colorSupplier.getRandomColor();
+
+        switch (figureType) {
+            case SQUARE:
+                double squareSideLength = random.nextDouble() * 5;
+                figure = Square.createSquare(squareSideLength, color);
+                break;

```

Comment:
I'm not a big fan of dangling null variables - instead of initializing nullable variable, you can return the figure immediately within the `case` block - that would also remove the need for `break` instruction.